File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
      Arrays.sort(arr);
        List<List<Integer>> ans = new ArrayList<>();
        int diff = Integer.MAX_VALUE;
        for (int i = 1; i < arr.length; ++i) {
            if (arr[i] - arr[i - 1] <= diff) {
                if (arr[i] - arr[i - 1] < diff) {
                    diff = arr[i] - arr[i - 1];
                    ans = new ArrayList<>();
                }
                ans.add(Arrays.asList(arr[i - 1], arr[i]));
            }
        }
        return ans;
    }
}

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
       
       sort(arr.begin(), arr.end());
        vector<vector<int>> res;
        int n = arr.size();
        int cur = INT_MAX;
        for (int i = 1; i < n; i++) {
            if (arr[i] - arr[i - 1] == cur) {
                res.push_back({arr[i - 1], arr[i]});
            }
            else if (arr[i] - arr[i - 1] < cur) {
                res = {{arr[i - 1], arr[i]}};
                cur = arr[i] - arr[i - 1];
            }
        }
        return res;
    }
};